src/compute/src/sink/subscribe.rs

```diff
@@ -89,92 +85,84 @@ fn subscribe<G>(
     as_of: Antichain<G::Timestamp>,
     up_to: Antichain<G::Timestamp>,
     subscribe_protocol_handle: Rc<RefCell<Option<SubscribeProtocol>>>,
-    probes: Vec<probe::Handle<Timestamp>>,
 ) where
     G: Scope<Timestamp = Timestamp>,
 {
-    // Let the subscribe sink emit a progress stream, so we can attach the flow control probes.
-    // The `Infallible` type signals that this steam never transports data updates.
-    // TODO: Replace `Infallible` with `!` once the latter is stabilized.
-    let progress_stream: Stream<G, Infallible>;
+    let name = format!("subscribe-{}", sink_id);
+    let mut op = OperatorBuilder::new(name, sinked_collection.scope());
+    let mut ok_input = op.new_input(&sinked_collection.inner, Pipeline);
+    let mut err_input = op.new_input(&err_collection.inner, Pipeline);
 
-    let mut rows_to_emit = Vec::new();
-    let mut errors_to_emit = Vec::new();
-    let mut finished = false;
-    let mut ok_buf = Default::default();
-    let mut err_buf = Default::default();
-    progress_stream = sinked_collection.inner.binary_frontier(
-        &err_collection.inner,
-        Pipeline,
-        Pipeline,
-        &format!("subscribe-{}", sink_id),
-        move |_cap, _info| {
-            move |ok_input, err_input, _output| {
-                if finished {
-                    // Drain the inputs, to avoid the operator being constantly rescheduled
-                    ok_input.for_each(|_, _| {});
-                    err_input.for_each(|_, _| {});
-                    return;
-                }
+    op.build(|_cap| {
```

Comment:
Probably not a huge one. I think it maybe reflects the intention of not producing an output a bit better, and I wasn't sure if it also avoids some amount of work caused by maintaining an output. It would be nice to have a `binary_sink` operator.

I have the opposite of a strong opinion, happy to change to `binary_frontier` if you prefer!